specs/builder.go

```diff
@@ -0,0 +1,216 @@
+package specs
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+	"unicode/utf8"
+
+	"github.com/moov-io/iso8583"
+	"github.com/moov-io/iso8583/encoding"
+	"github.com/moov-io/iso8583/field"
+	"github.com/moov-io/iso8583/padding"
+	"github.com/moov-io/iso8583/prefix"
+)
+
+const (
+	FieldPrefix = "Field"
+)
+
+var (
+	fieldConstructor = map[string]func(spec *field.Spec) field.Field{
```

Comment:
@mfdeveloper508 sorry, but I forgot that we can use function as a type and change the map into this:

```go
type fieldConstructorFunc func(spec *field.Spec) field.Field

fieldConstructor = map[string]fieldConstructorFunc{
"String": field.NewString,
"Numeric": field.NewNumeric,
// ...
}
```